src/app/rundown/components/rundown/rundown.component.ts

```diff
@@ -16,7 +16,10 @@ export class RundownComponent implements OnInit, OnDestroy {
 
   public rundown?: Rundown
   public adLibPieceIdentifiers: Identifier[] = []
+  public isLoading: boolean = true
```

Comment:
Your current implementation does not consider if we are done loading and we didn't get a Rundown (which is fine).
But since you only have two states "I don't have a Rundown" and "I have a Rundown", then you don't need this `isLoading` variable because `rundown` is optional.

As long as `rundown` is undefined, you show your loading message. When it's no longer undefined, you show the Rundown.

Reply:
If you implemenet it like this, then you don't need to subscribe to the loading subject and then you don't need to implement the loading subject in the state service at all (atleast for now).

Reply:
But it does also mean we at this point in time wouldn't be able to show a message saying "This Rundown has been deleted" etc...

Reply:
I have changed this to the much simpler suggestion!